contracts/oracle/src/contract/alarms.rs

```diff
@@ -25,14 +26,23 @@ use crate::{
 
 pub type AlarmReplyId = u64;
 
-pub struct MarketAlarms;
+pub struct MarketAlarms(AlarmReplyId);
 
 impl MarketAlarms {
     const PRICE_ALARMS: PriceAlarms<'static> =
         PriceAlarms::new("alarms_below", "index_below", "alarms_above", "index_above");
```

Comment:
Create a new instance always and store it in MarketAlarms. Pass a storage instance. Then a lot of code will become obsolete and be removed, e.g. passing a storage down the stack

Reply:
fixed with #22